snapcraft/integrations/travis.py

```diff
@@ -0,0 +1,183 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2015 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Snapcraft integration for Travis (CI).
+
+This is an *EXPERIMENTAL* feature and subject to incompatible changes in
+the future, please use with caution.
+
+This command currently depends on a working `travis` CLI environment and
+a previously initialized Travis project (`.travis.yml`).
+
+Make sure your Travis project is also configured to "Build pushes", this
+way every new push to `master` will result in a new snap revision in the
+Store.
+
+This operation will acquire properly attenuated Store credentials and
+encrypt them for use in your testbed (`.travis_snapcraft.cfg`), only Travis
+has the private key to decrypt it and will be only available to branches
+of the same repository, not forks.
+
+Then it will adjust Travis configuration ('.travis.yml') with the commands
+to decrypt credentials during 'after_success' phase and install latest
+`snapcraft` to build and release your snap (inside a ubuntu:xenial docker
+container) during the 'deploy' phase.
+
+See the example below::
+
+    sudo: required
+    services:
+    - docker
+    after_success:
+    - openssl aes-256-cbc -K <travis-key> -iv <travis-iv>
+      -in .travis_snapcraft.cfg -out .snapcraft.cfg -d
```

Comment:
One more detail, when this is done from a fork, the encripted env vars will not be set. So it prints:
iv undefined.
It would be nice to do the decrypt only in master, too.